.github/workflows/analysis.yml

```
@@ -0,0 +1,177 @@
+        run: make test-unit
+
+      - name: Upload coverage for SonarQube Cloud
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage
+          path: reports/coverage.xml
+          retention-days: 1
+
+      - name: Save PR metadata
+        if: github.event_name == 'pull_request'
+        run: |
```

actionlint (1.7.11)

[error] "github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)